tests/test_coordinator.py

```diff
@@ -564,3 +565,71 @@ def test_post_aggregation_fails(
     assert fee_model.totalPendingFees() == 0
     assert fee_model.pendingFees(ritualID) == 0
     fee_model.withdrawTokens(fee_model_balance_after_refund, sender=deployer)
+
+
+def test_upgrade(
+    coordinator, nodes, initiator, erc20, fee_model, treasury, deployer, global_allow_list
+):
+    coordinator.initiateOldRitual(
+        fee_model, nodes, initiator, DURATION, global_allow_list.address, sender=initiator
+    )
+    coordinator.initiateOldRitual(
+        ZERO_ADDRESS, [nodes[0]], treasury, DURATION // 2, deployer, sender=initiator
+    )
+    assert coordinator.numberOfRituals() == 0
+    coordinator.initializeNumberOfRituals(sender=deployer)
+    assert coordinator.numberOfRituals() == 2
+
+    initiate_ritual(
+        coordinator=coordinator,
+        fee_model=fee_model,
+        erc20=erc20,
+        authority=initiator,
+        nodes=nodes,
+        allow_logic=global_allow_list,
+    )
+    assert coordinator.numberOfRituals() == 3
+
+    assert coordinator.getRitualState(0) == RitualState.DKG_AWAITING_TRANSCRIPTS
+
+    ritual_struct = coordinator.rituals(0)
+    assert ritual_struct[0] == initiator
+    init, end = ritual_struct[1], ritual_struct[2]
+    assert end - init == DURATION
+    total_transcripts, total_aggregations = ritual_struct[3], ritual_struct[4]
+    assert total_transcripts == total_aggregations == 0
+    assert ritual_struct[5] == initiator
+    assert ritual_struct[6] == len(nodes)
+    assert ritual_struct[7] == 1 + len(nodes) // 2  # threshold
+    assert not ritual_struct[8]  # aggregationMismatch
+    assert ritual_struct[9] == global_allow_list.address  # accessController
+    assert ritual_struct[10] == (b"\x00" * 32, b"\x00" * 16)  # publicKey
+    assert not ritual_struct[11]  # aggregatedTranscript
```

Comment:
If I'm understanding this test correctly, you are creating and checking 3 rituals:
1. first one mimics ritual that was created before any concept of fee model - fee model is 0x...0
2. second mimics ritual that was created with a fee model address but was initiated with the storage before your current storage change fix
3. third mimics ritual that was created with your current storage change fix

For `v7.5.x` (so unreleased code for next release) we use index 12 for fee model - https://github.com/nucypher/nucypher/commit/2c2cb5dc05f0650d5f95bf5a4042dd262849c06f#diff-911a370c18d0d0e3ea62099fc5ff6daa16915da19cf5399f24e404b7d8d6d168R597. 
- Is that still fine or will that be problematic?
- If not problematic, should we also check the values in the struct at index `12` (fee model) in this test?

Reply:
for client that supports `new fee model architecture` - everything should stay the same, so yeah feeModel is 12th in structure.
all add test in check

Reply:
added this check
interesting that `ritual_struct[12]` is participants array and not fee model
I'm not sure if it's ape behavior or abi enhanced for nested structs comparing to what was before, anyway could you check if 7.5.x would work with this update, please

Reply:
> interesting that ritual_struct[12] is participants array and not fee model

Indeed. This was how we realized we needed to get participants for a ritual in a separate call.

> could you check if 7.5.x would work with this update, please

Ran the acceptance tests for `v7.5.x` with the following update to the ape-config.yaml:
```diff
diff --git a/tests/acceptance/ape-config.yaml b/tests/acceptance/ape-config.yaml
index b44995ae6..3d3d34ce2 100644
--- a/tests/acceptance/ape-config.yaml
+++ b/tests/acceptance/ape-config.yaml
@@ -5,8 +5,8 @@ plugins:

 dependencies:
   - name: nucypher-contracts
-    github: nucypher/nucypher-contracts
-    ref: main
+    github: vzotova/nucypher-contracts
+    ref: coordinator-layout
     config_override:
       solidity:
         version: 0.8.23
```

and the acceptance tests all pass without any changes 🎉 .